onnxruntime/contrib_ops/webgpu/quantization/dp4a_matmul_nbits.cc

```diff
@@ -258,11 +258,120 @@ Status DP4AMatMulNBitsProgram::GenerateShaderCode(ShaderHelper& shader) const {
   return Status::OK();
 }
 
+// tile_N size = 16, workgroup size = 64, scale_A components = 1, b components = 4, output components = 4
+Status DP4AMatMulNBitsSmallMProgram::GenerateShaderCode(ShaderHelper& shader) const {
+  shader.AddInput("input_a", ShaderUsage::UseUniform);
+  shader.AddInput("scales_a", ShaderUsage::UseUniform);
+  shader.AddInput("input_b", ShaderUsage::UseUniform);
+  shader.AddInput("scales_b", ShaderUsage::UseUniform);
+  shader.AddOutput("output", ShaderUsage::UseUniform | ShaderUsage::UseElementTypeAlias);
+
+  shader.AdditionalImplementation() << R"ADDNL_FN(
+    const tile_size = 16u; // tile_size = tile_size_vec * output components
+    const tile_size_vec = 4u;
+    const tile_size_k_vec = 16u; // tile_size_vec * tile_size_k_vec = workgroup size
+    // Shared memory
+    var<workgroup> tile_A : array<vec4<u32>, 32>;                    // 512 scalars
+    var<workgroup> scale_A : array<output_element_t, 4>;             // 4
+    var<workgroup> inter_results: array<array<vec4<output_element_t>, tile_size_k_vec>, tile_size_vec>;
+    fn loadSHMA(a_global:u32, kidx_v:u32, col: u32)
+    {
+      let k_offset = kidx_v + col;
+      if (k_offset >= uniforms.K16) {
+        return;
+      }
+
+      tile_A[col] = input_a[a_global*uniforms.K16+k_offset];
+      if (col < 4)
+      {
+        // kidx_v - covers 16 values of k
+        scale_A[col] = scales_a[a_global*(uniforms.K/128) + k_offset/8 + col];
```

Comment:
scales_a[a_global*(uniforms.K/128) + k_offset/8 + col];

I think this should be 

scales_a[a_global*(uniforms.K/128) + k_offset/8];

because k_offset already contains col.

Reply:
It should be `scale_A[col] = scales_a[a_global*(uniforms.K/128) + kidx_v/8 + col];`. I need 4 scale a's data.